promises_intro/index.js

```diff
@@ -0,0 +1,89 @@
+
+const addPostForm = document.getElementById('add-post-form');
+const addPostInput = document.getElementById('add-post-input');
+const deletePostForm = document.getElementById('delete-post-form');
+const deletePostInput = document.getElementById('delete-post-input');
+const ul = document.querySelector("ul");
+const cachedPosts = [];
+
+
+const addPost = (post) => {
+    const li = document.createElement("li");
+    const { title: postTitle } = post;
+    li.innerText = postTitle;
+    ul.append(li);
+}
+
+const getPosts = fetch("http://localhost:3000/posts");
+
+getPosts.then((response) => response.json())
+        .then((posts) => {
+            posts.forEach((post) => {
+                addPost(post);
+                cachedPosts.push(post);
+            })
+        })
+
+addPostForm.addEventListener('submit', (event) => {
+    event.preventDefault();
+    
+    const title = addPostInput.value;
+    const id = String(Math.random()); 
+
+    //check if post with this title exists
+    if (cachedPosts.find((post) => post.title === title)) {
+        alert(`Post with title ${title} already exists!`);
+    } else {
```

Comment:
you can return before "else" and skip else condition (just continue with flow without else)